firmware/quadruna/VC/src/app/vehicle_dynamics/app_pid.c

```diff
@@ -22,8 +25,23 @@ void app_pid_init(PID *pid, const PID_Config *conf)
 float app_pid_compute(PID *pid, float setpoint, float input)
 {
     pid->error = setpoint - input;
-    pid->integral += pid->error;
-    pid->derivative = (input - pid->prev_input);
+
+    if(app_gaurd_init(pid)){ // if initialized implement the gaurd
+        if(pid->integral_windup_max > pid->integral && pid->integral > pid->integral_windup_min){
+            pid->integral += pid->error;
+        }
+    } 
+    else{
+        pid->integral += pid->error;
+    }
+
+    if(app_filter_init(pid)){
```

Comment:
i dont think this is the proper low pass filtering you're trying to achieve since it doesnt use the derivatives history. ur calculating this based on input rn not error. I don't rly know it at the top of my head rn, but just gpt it or something